apis/management.cattle.io/v3/logging_types.go

```diff
@@ -87,47 +87,35 @@ type LoggingCondition struct {
 }
 
 type ElasticsearchConfig struct {
-	Host         string `json:"host,omitempty"`
-	Port         int    `json:"port,omitempty"`
-	IndexPrefix  string `json:"indexPrefix,omitempty"`
-	DateFormat   string `json:"dateFormat,omitempty"`
+	Endpoint     string `json:"endpoint,omitempty" norman:"required"`
+	IndexPrefix  string `json:"indexPrefix,omitempty" norman:"required"`
+	DateFormat   string `json:"dateFormat,omitempty" norman:"required,type=enum,options=YYYY-MM-DD|YYYY-MM|YYYY,default=YYYY-MM-DD"`
 	AuthUserName string `json:"authUsername,omitempty"` //secret
 	AuthPassword string `json:"authPassword,omitempty"` //secret
 }
 
 type SplunkConfig struct {
-	Host     string `json:"host,omitempty"`
-	Port     int    `json:"port,omitempty"`
-	Protocol string `json:"protocol,omitempty"`
+	Endpoint string `json:"endpoint,omitempty" norman:"required"`
```

Comment:
What is the format for the endpoint? Is it meant as a replacement for address/port/protocol? If yes, is there any reason why we don't use a composite field for that, something similar to:

https://github.com/rancher/types/blob/d6b26cedb92786323fb79adf4af529d927366fd7/apis/cluster.cattle.io/v3/schema/schema.go#L71

Reply:
I see in the 1.6 the publicEndpoint like this, still the same?
```
"publicEndpoints": [ 
{
"type": "publicEndpoint",
"hostId": "1h7",
"instanceId": "1i510503",
"ipAddress": "120.76.202.180",
"port": 8081,
"serviceId": "1s629"
},
{
"type": "publicEndpoint",
"hostId": "1h7",
"instanceId": "1i526194",
"ipAddress": "120.76.202.180",
"port": 500,
"serviceId": "1s5"
}]
```

current the public endpoint for different target I expected user may input different format, for example: elasticsearch: https://ip:port, syslog: ip:port.(because it use udp/tcp protocol)

Reply:
@aiwantaozi thats fine, spoke with @vincent99, from the UI perspective we are good having endpoint as a string.